### PR TITLE
[Component with its own class name as service type] LPS-198392 Refactor components from batch-planner-rest-impl and portal-search-solr8-impl modules 

### PR DESCRIPTION
**Pull request by RenanUmV**
[LPS-198392](https://liferay.atlassian.net/browse/LPS-198392)

Previous Pull: https://github.com/rafaprax/liferay-portal/pull/1331

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
ci:test:search

**Comment by catarineaguiar** [comment]
no regressions found

**Review by rafaprax — APPROVED** [review]

File: modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/portlet/AssetPublisherPortlet.java [hunk 1]
```diff
@@ -490,9 +510,170 @@ protected boolean isSessionErrorException(Throwable throwable) {
...
+
+			syndEntry.setPublishedDate(assetEntry.getPublishDate());
+			syndEntry.setTitle(assetEntry.getTitle(languageId, true));
+			syndEntry.setUpdatedDate(assetEntry.getModifiedDate());
+			syndEntry.setUri(link);
+
+			syndEntries.add(syndEntry);
+		}
+
+		syndFeed.setFeedType(RSSUtil.getFeedType(format, version));
```
Comment (rafaprax) [inline modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/portlet/AssetPublisherPortlet.java:576]: Hey @MrJohn1911, I know it was not included by you but try to remove the not needed declared variables.

File: modules/apps/batch-planner/batch-planner-rest-impl/src/main/java/com/liferay/batch/planner/rest/internal/resource/v1_0/FieldResourceImpl.java [hunk 2]
```diff
@@ -33,17 +36,19 @@ public Page<Field> getPlanInternalClassNameKeyFieldsPage(
 		throws Exception {
 
 		List<com.liferay.portal.vulcan.batch.engine.Field> vulcanFields =
-			_fieldProvider.getFields(
+			FieldProviderUtil.getFields(
```
Comment (rafaprax) [inline modules/apps/batch-planner/batch-planner-rest-impl/src/main/java/com/liferay/batch/planner/rest/internal/resource/v1_0/FieldResourceImpl.java:39]: Hey @MrJohn1911, try to use the Base class approach here.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by RenanUmV** [comment]
resent here: https://github.com/liferay-core-infra/liferay-portal/pull/6158

**Pull request closed without merging**